libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2508,29 +2507,56 @@ string YulUtilFunctions::revertReasonIfDebug(RevertStrings revertStrings, string
 				mstore(add(reasonPos, <offset>), <wordValue>)
 			</word>
 			revert(0, add(reasonPos, <end>))
-		})");
-		templ("sig", (u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")))) << (256 - 32)).str());
+		<!debugAndMessage>
+			revert(0, 0)
+		</debugAndMessage>
+	})");
+	bool debugAndMessage = _revertStrings >= RevertStrings::Debug && !_message.empty();
+	templ("debugAndMessage", debugAndMessage);
+	if (debugAndMessage)
+	{
+		templ("sig", (u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)"))))
+				<< (256 - 32)).str());
 		templ("length", to_string(_message.length()));
 
 		size_t words = (_message.length() + 31) / 32;
 		vector<map<string, string>> wordParams(words);
-		for (size_t i = 0; i < words; ++i)
-		{
+		for (size_t i = 0; i < words; ++i) {
 			wordParams[i]["offset"] = to_string(i * 32);
 			wordParams[i]["wordValue"] = formatAsStringOrNumber(_message.substr(32 * i, 32));
 		}
 		templ("word", wordParams);
 		templ("end", to_string(words * 32));
+	}
+
+	return templ.render();
+}
+
+string YulUtilFunctions::revertReasonIfDebugFunction(RevertStrings _revertStrings, string const& _message)
+{
+	string functionName = "revert_reason_" + revertStringsToString(_revertStrings) + "_";
+	for (char c: _message)
+		if (isalnum(c))
+			functionName += tolower(c);
+		else if (isspace(c))
+			functionName += "_";
```

Comment:
Is message `_message` guaranteed to contain only ASCII characters? Or can the message be UTF-8? If the latter then using `isalnum()` and `tolower()` is not a good idea since `c` might not be a whole character.

Reply:
Hm, you are right.
Do you have any suggestion how to name those functions in a better way?
I did propose this as I did not have better ideas :)

Reply:
Maybe you could just hash the message? Then unicode characters would not be a problem.

You could also combine this with the message stripped of any non-ASCII characters (so basically only chars < 0x8f). This would make the name easier to recognize in case it does contain any ASCII characters. It might not if it's in Chinese or is entirely made up of emoji :)